feat(lanelet2_extension): add multi osm parser

StepTurtle opened the pull request:
## Description

I have added a new library class to facilitate the loading of multiple OSM files in lanelet2_extension. These changes were made to support dynamic lanelet loading.

## Related links

[Proposal Link](https://github.com/orgs/autowarefoundation/discussions/4120)

## Tests performed

In this video, the map in background loaded with current approach and the white map load new class and cannot see any difference. Also the maps which loaded with new class tested with mission and behavior planner and cannot see any problem.

[dynamic_load_map.webm](https://github.com/autowarefoundation/autoware_common/assets/76053179/9f81d55e-e0a8-4cec-b254-aaaa3fb78042)


## Notes for reviewers

## Interface changes

## Effects on system behavior

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

esteve reviewed (changes requested):
@StepTurtle is the code in the `tmp` folder meant to be included in this PR? There also seems to be changes unrelated to this PR that are copied from https://github.com/autowarefoundation/autoware_common/pull/238, is that intended?

Commit:
feat(lanelet2_extension): add multi osm parser

Signed-off-by: M. Fatih Cırıt <mfc@leodrive.ai>

StepTurtle commented:
> @StepTurtle is the code in the `tmp` folder meant to be included in this PR? There also seems to be changes unrelated to this PR that are copied from #238, is that intended?

Hi @esteve 
When you said this, I realized I had added some wrong parts to the PR. Now we fixed everything and it looks correct. I also apologize for any inconvenience I may have caused. Thank you for your warning
